pkg/controllers/status/common.go

```diff
@@ -105,43 +106,53 @@ func updateResourceStatus(
 ) error {
-	gvr, err := restmapper.GetGroupVersionResource(restMapper, schema.FromAPIVersionAndKind(resource.GetAPIVersion(), resource.GetKind()))
+	gvr, err := restmapper.GetGroupVersionResource(restMapper, schema.FromAPIVersionAndKind(objRef.APIVersion, objRef.Kind))
 	if err != nil {
-		klog.Errorf("Failed to get GVR from GVK(%s/%s), Error: %v", resource.GetAPIVersion(), resource.GetKind(), err)
+		klog.Errorf("Failed to get GVR from GVK(%s/%s), Error: %v", objRef.APIVersion, objRef.Kind, err)
 		return err
 	}
 
-	if !interpreter.HookEnabled(resource.GroupVersionKind(), configv1alpha1.InterpreterOperationAggregateStatus) {
+	gvk := schema.GroupVersionKind{Group: gvr.Group, Version: gvr.Version, Kind: objRef.Kind}
+	if !interpreter.HookEnabled(gvk, configv1alpha1.InterpreterOperationAggregateStatus) {
 		return nil
 	}
+
+	// get resource template directly from apiserver, rather than informer cache, detail refers to:
+	// https://github.com/karmada-io/karmada/issues/5285
+	resource, err := dynamicClient.Resource(gvr).Namespace(objRef.Namespace).Get(ctx, objRef.Name, metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			// It might happen when the resource template has been removed but the garbage collector hasn't removed
+			// the ResourceBinding which dependent on resource template.
+			// So, just return without retry(requeue) would save unnecessary loop.
+			return nil
+		}
+		klog.Errorf("Failed to fetch resource(%s/%s) for resourceBinding. Error: %v.", resource.GetNamespace(), resource.GetName(), err)
+		return err
+	}
+
 	newObj, err := interpreter.AggregateStatus(resource, bindingStatus.AggregatedStatus)
 	if err != nil {
-		klog.Errorf("Failed to aggregate status for resource(%s/%s/%s, Error: %v", gvr, resource.GetNamespace(), resource.GetName(), err)
+		klog.Errorf("Failed to aggregate status for resource(%s/%s/%s), Error: %v", gvr, resource.GetNamespace(), resource.GetName(), err)
 		return err
 	}
 
-	oldStatus, _, _ := unstructured.NestedFieldNoCopy(resource.Object, "status")
-	newStatus, _, _ := unstructured.NestedFieldNoCopy(newObj.Object, "status")
-	if reflect.DeepEqual(oldStatus, newStatus) {
+	if reflect.DeepEqual(resource, newObj) {
 		klog.V(3).Infof("Ignore update resource(%s/%s/%s) status as up to date.", gvr, resource.GetNamespace(), resource.GetName())
 		return nil
 	}
 
-	patchBytes, err := helper.GenReplaceFieldJSONPatch("/status", oldStatus, newStatus)
-	if err != nil {
-		klog.Errorf("Failed to gen patch bytes for resource(%s/%s/%s, Error: %v", gvr, resource.GetNamespace(), resource.GetName(), err)
-		return err
-	}
-
-	_, err = dynamicClient.Resource(gvr).Namespace(resource.GetNamespace()).
-		Patch(ctx, resource.GetName(), types.JSONPatchType, patchBytes, metav1.PatchOptions{}, "status")
-	if err != nil {
+	if _, err = dynamicClient.Resource(gvr).Namespace(resource.GetNamespace()).UpdateStatus(ctx, newObj, metav1.UpdateOptions{}); err != nil {
```

Comment:
I'm not very clear, why would using update be less prone to conflicts than using patch?

Reply:
Previously, the root cause of conflicts is that we get RT (resource template) from cache, and that cache may not be updated in time (Now we get RT directly from apiserver to reduce the probability conflicts).

The number of conflicts has nothing to do with using `update` or `patch`, the difference is only that `patch` ignore conflict, while `update` will retry on conflict.

I replace `patch` with `update` is just I think `update` is more elegant, for `patch` will let apiserver always write etcd, while `update` will avoid write when `resourceVersion` is equal.